### PR TITLE
JD-2 gh-34 add proxy support

### PR DESCRIPTION
**Pull request by geraltrivyan**
https://github.com/jsdelivr/jsdelivr-log-parser/issues/34

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.idea/
```
Comment (MartinKolarik) [inline .gitignore]: This isn't ignored on purpose. If you run into issues with it please explain on Slack.

File: src/models/Proxy.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		this.id = 0;
+
+		/** @type {string} */
+		this.path = '';
+
+		Object.assign(this, properties);
+		return new Proxy(this, BaseModel.ProxyHandler);
+	}
+
+	static async getProxyList () {
```
Comment (MartinKolarik) [inline src/models/Proxy.js]: Probably just use `.findAll()` which is inherited from `BaseModel` ?

**Comment by geraltrivyan** [comment]
https://github.com/jsdelivr/jsdelivr-log-parser/issues/34

File: src/models/Proxy.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+const Joi = require('joi');
+const BaseModel = require('./BaseModel');
+
+const schema = {
+	id: Joi.number().integer().min(1).required().disallow(null),
```
Comment (MartinKolarik) [inline src/models/Proxy.js]: I think `.disallow(null)` is not needed anywhere, it shouldn't accept null by default.

File: src/models/ProxyHits.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+const Joi = require('joi');
+const BaseModel = require('./BaseModel');
+
+const schema = {
+	id: Joi.number().integer().min(1).required().allow(null),
```
Comment (MartinKolarik) [inline src/models/ProxyHits.js]: Remove and use (`proxyId`, `date`) as the primary key (everything should work like in `FileHits`).

File: src/models/Proxy.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	path: Joi.string().required(),
```
Comment (MartinKolarik) [inline src/models/Proxy.js]: `Joi.string().max(255).required()`

File: src/models/Proxy.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+	static get unique () {
+		return [ 'id' ];
+	}
+
+	constructor (properties = {}) {
+		super();
+
+		/** @type {number} */
+		this.id = 0;
```
Comment (MartinKolarik) [inline src/models/Proxy.js]: `null`

File: src/models/ProxyHits.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+const Joi = require('joi');
+const BaseModel = require('./BaseModel');
+
+const schema = {
+	proxyId: Joi.number().integer().min(1).required().allow(null),
```
Comment (MartinKolarik) [inline src/models/ProxyHits.js]: Remove `allow(null)`

**Pull request merged**